pkg/autodiscovery/configresolver.go

```diff
@@ -187,7 +184,8 @@ func (cr *ConfigResolver) resolve(tpl integration.Config, svc listeners.Service)
 	}
 
 	// store resolved configs in the AC
-	cr.ac.loadedConfigs = append(cr.ac.loadedConfigs, resolvedConfig)
+	cr.ac.loadedConfigs[resolvedConfig.Digest()] = resolvedConfig
+	cr.ac.store.addConfigForService(svc.GetID(), resolvedConfig)
```

Comment:
I don't think the config resolver should know about the autodiscovery. There is a dependency cycle here which breaks the responsibility principle. I'd rather do all those operations upstream in autodiscovery or just inject the `store` to the config resolver.

Reply:
This will be solved once the service listening is moved away from the config resolver